src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -44,6 +44,19 @@ def event_activate(self):
             f'{self.channel_prefix}/device_info': self._device_info,
         })
 
+    def _is_ess_buy_limit_surpassed(self, energy):
+        return energy > self.state.energy_to_buy_dict[self.market.time_slot]
+
+    def _is_ess_sell_limit_surpassed(self, energy):
+        return energy > self.state.energy_to_sell_dict[self.market.time_slot]
+
+    def _publish_error_message_to_response_channel(self, response_channel, transaction_id):
```

Comment:
This function is only for error messages concerning the ESS limit.
Please either rename the function or transform it into an even more generic function that takes the command, error and transaction_id as input.
I guess that the first will be sufficient though.

Reply:
@hannesdiedrich I would prefer if the old way was retained where an assertion is raised, and caught afterwards. That way we are sure that in the impl method there is only one publish_json for an error condition, which makes the code more uniform.

Reply:
You are right, agreed!

Reply:
Sorry, now im confused, before my recent commit, I was using an if clause and publishing message to response channel, I wasn't using `assert` for the failure of `try except block`.

Reply:
Discussed offline all the state validation will be moved to `_impl` method and keeping generic error message